src/components/Categories/Categories.js

```diff
@@ -0,0 +1,78 @@
+import {useEffect, useState} from 'react';
+import styled from 'styled-components';
+
+export default function Categories({menus, handleCategoryClick, curClickedCategory}) {
+  const [categories, setCategories] = useState([]);
+
+  useEffect(() => {
+    /* Get categories form the given menus */
+    function getCategories(menus) {
+      const categories = menus.reduce(
```

Comment:
I think reduce is not the right method for this case. Maybe there would be a other solution? What is your intention for this code?

Reply:
function getAlternativeCategorie(menus) {
  const categories = ["All"]
  menus.forEach(menu => {
    if (!categories.includes(menu.category)) {
      categories.push(menu.category)
    }
  })
  return categories
}

Reply:
const sortBy = ["All", "Starters", "Mains", "Pastries", "Drinks"];

function getCategoriesAndSort(menus, sortBy) {
  const categories = ["All"];
  menus.forEach((menu) => {
    if (!categories.includes(menu.category)) {
      categories.push(menu.category);
    }
  });
  
  let sortByObj = {};
  sortBy.forEach((menuPoint, index) => {
    sortByObj = { ...sortByObj, [menuPoint]: index };
  });

  const sortedCategories = categories.sort(
    (a, b) => sortByObj[a] - sortByObj[b]
  );

  return sortedCategories;
}